tubular/scripts/prune_modulestore.py

```
@@ -0,0 +1,559 @@
+#! /usr/bin/env python3
+
+"""
+[WIP]
+
+Command-line script used to clean up (trim) the modulestore structure. By virtue of its nature, the module store
+versions updates and over a period of time, these updates account for a significant growth in the size of the
+mongo database.
+
+This script prunes the modulestore structures using the parameters specified.
+
+The final product will support:
+    1. dry-run:
+    2. prune targeted course/active version
+    3. prune all active versions
+    4. support tests via static data
+    5. visualize targeted course trees
+
+Options 2 & 3 support removing all structures or keeping a number of older structures (in support of
+user-specified retention policy)
+
+See more details regarding module store at
+http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/modulestores/split-mongo.html
+
+See additional details regarding the growth problem with the modulestore at
+https://openedx.atlassian.net/browse/PLAT-697
+
+"""
+
+from __future__ import absolute_import
+
+import json
+import logging
+import os
+import sys
+import click
+import click_log
+from bson.objectid import ObjectId
+from pymongo import MongoClient
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+LOG = logging.getLogger(__name__)
+
+# the dictionary keys to track for active versions
+TARGET_ACTIVE_VERSIONS_KEYS = [u'library', u'draft-branch', u'published-branch']
```

Okay. I think conservative is fine.

1. Those are the only ones we expect. Maybe we can just log if we find a key that we don't anticipate? That way if we do add a new type later on, we'll at least log something so we don't silently skip pruning -- while at the same time preserving the conservative behavior.
2. There is no documentation that I'm aware of aside from the link to source code comments I already mentioned.
3. If you mean for the three types already listed, I can test locally on my devstack if you'd like to help verify, but I'm guessing you folks have already done that for those three. If not, I can do that on Friday (I'm really booked up tomorrow).

Thank you!